src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2192,19 +2180,8 @@ private void onDecodeTransfer(
                     boolean batchable = transfer.hasBatchable() && transfer.batchable() == 1;
                     boolean more = transfer.hasMore() && transfer.more() == 1;
                     AmqpServerStream link = links.get(transfer.handle());
-                    if (link.deliveryId == -1 || !transfer.hasMore() || transfer.more() == 0)
-                    {
-                        link.linkCredit--;
-                    }
-                    if (link.linkCredit < 0)
-                    {
-                        link.onDecodeError(traceId, authorization, LINK_TRANSFER_LIMIT_EXCEEDED);
-                    }
-                    else
-                    {
-                        link.onDecodeTransfer(traceId, authorization, reserved, deliveryId, deliveryTag, messageFormat, settled,
-                            resume, aborted, batchable, more, buffer, offset, limit);
-                    }
+                    link.onDecodeTransfer(traceId, authorization, reserved, deliveryTag, messageFormat, settled,
+                        resume, aborted, batchable, more, buffer, offset, limit);
```

Comment:
Do we need to keep these checks to validate inbound `deliveryId` or is it fully handled inside the link instead now?

Reply:
I believe I handled it inside the link's `onDecodeTransfer`